tests/integration/pilot/ingress_test.go

```diff
@@ -728,6 +733,9 @@ resources:
     cpu: 10m
     memory: 40Mi
 `, rev)), 0o644)
+				if t.Settings().OpenShift {
+					_ = file.AppendToFile([]byte(`platform: openshift`), d)
```

Comment:
as we're writing the file in the line above - wouldn't it be more efficient to append to the string before writing?

Reply:
I considered that, but I felt that using `file.AppendToFile` was cleaner and easier to read, so I chose this approach. I'm open to changing it if that's preferred.